src/ring_theory/localization.lean

```diff
@@ -1335,6 +1382,21 @@ noncomputable def localization_algebra (M : submonoid R) (f : localization_map M
 variables (f : localization_map M Rₘ)
 variables (g : localization_map (algebra.algebra_map_submonoid S M) Sₘ)
 
+/-- Injectivity of the underlying `algebra_map` descends to the algebra induced by localization -/
+lemma localization_map_injective (hRS : function.injective (algebra_map R S))
+  (hM : (algebra.algebra_map_submonoid S M) ≤ non_zero_divisors S) :
+  function.injective (@algebra_map Rₘ Sₘ _ _ (localization_algebra M f g)) :=
+begin
+  rintros x y hxy,
+  obtain ⟨a, b, hx⟩ := localization_map.mk'_surjective f x,
+  obtain ⟨c, d, hy⟩ := localization_map.mk'_surjective f y,
+  rw [← hx, ← hy] at hxy ⊢,
+  iterate 2 {erw localization_map.map_mk' at hxy},
```

Comment:
Could you extract a lemma `localization_algebra_mk' : (algebra_map Rₘ Sₘ) (f.mk' a b) = g.mk' (⇑(algebra_map R S) a) ⟨⇑(algebra_map R S) ↑b, _⟩`? Then this can be a normal `rw`.